broadinstitute/single_cell_portal_core
# Sparse matrices, features file, barcodes file basic client-side validation (SCP-3766)

Add basic client-side validation for Sparse Matrices, Features files and Barcodes files. This does not cover the validation that can be done for Features and Barcodes using info from the MTX file. Only intrafile validation.

To Test:
Pull this branch
go to the upload wizard
choose Sparse matrix for the file type
try to upload a sparse matrix that has: A blank line, comment lines, missing the header line '%%MatrixMarket', etc.
try to upload a Features file with a duplicate 
try to upload a Barcodes file with a duplicate 

note the duplication check is following the way ingest pipeline does, where it looks at the entire row as a value that can be duplicated, not at the individual pieces of a row.

![Screen Shot 2022-01-13 at 4 39 05 PM](https://user-images.githubusercontent.com/54322292/149415981-dbb1bb53-32da-45ae-92ca-bf0b177ac0e6.png)
![Screen Shot 2022-01-11 at 4 03 07 PM](https://user-images.githubusercontent.com/54322292/149416319-37bc6f93-659f-444b-b8dc-2701447e8ac0.png)
